pkg/controllers/gateway/gateway_controller.go

```diff
@@ -351,6 +375,99 @@ func selectClusters(gateway gatewayv1beta1.Gateway) []string {
 	return []string{}
 }
 
+func (r *GatewayReconciler) DNSDeletion(ctx context.Context, gateway gatewayv1beta1.Gateway) error {
+	trafficAccessor := traffic.NewGateway(&gateway)
+	allHosts := trafficAccessor.GetHosts()
+
+	for _, host := range allHosts {
+		managedZone, subDomain, err := r.Host.GetManagedZoneForHost(ctx, host, trafficAccessor)
+		if err != nil {
+			return err
+		}
+		if managedZone == nil {
+			log.Log.Info("Managed zone not found cannot get DNS record")
+			return nil
+		}
+
+		log.Log.Info("getting DNS record associated with deleting gateway")
+		dnsRecord, err := r.Host.GetDNSRecord(ctx, subDomain, managedZone)
+		if err != nil && !k8serrors.IsNotFound(err) {
+			return err
+		}
+		if dnsRecord == nil {
+			log.Log.Info("dns record not found")
```

Comment:
perhaps more useful if we log the host with it when not found?

Reply:
Sure